src/main/scala/scala/collection/IteratorExtensions.scala

```diff
@@ -0,0 +1,40 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.collection
+
+object IteratorExtensions {
+  implicit class IteratorExtensionsOps[A](private val iter: Iterator[A]) extends AnyVal {
```

Comment:
any reason this is an extension to `Iterator` and not to `IterableOnceOps`?

Reply:
`Iterable` also inherits `IterableOnceOps`, but it already has `groupMapReduce`. Do we have another type than `Iterator` that would benefit from the operation?

Reply:
I was about to ask the same, what would be the benefit of adding this method to any **IterableOnce**?

Reply:
theoretically someone else can add their own `IterableOnceOps` implementation and might want it

Reply:
@NthPortal ok that is a valid argument, but one question, why `IterableOnceOps`? Shouldn't it be `IterableOnce`?

Reply:
the other reason for putting it in `IterableOnceOps` is that that will be the eventual location in 3.1, and there won't be two copies of the method

--------

> Why `IterableOnceOps`? Shouldn't it be `IterableOnce`

no. @julienrf can probably answer this better than I can, but essentially, `IterableOnce` is meant to be barebones, and not contain all of the transformations